src/ros_filter.cpp

```diff
@@ -2045,6 +2205,29 @@ namespace RobotLocalization
     }
 
     ros::Time curTime = ros::Time::now();
+    double secCurTime = curTime.toSec();
+
+    // Handle any timeouts of trusted data
+    if((trusted_timeout_ > 0.0) || (bias_timeout_ > 0.0))
+    {
+      for(auto &item : sourceData_)
+      {
+        if((trusted_timeout_ > 0.0) && 
+          (item.second.trusted_) &&
+          (secCurTime > (item.second.last_trusted_s_ + trusted_timeout_)))
+        {
+          // Trusted status timed out
+          item.second.trusted_ = false;
+        }
+        if((bias_timeout_ > 0.0) && 
+          (item.second.bias_valid_) &&
+          (secCurTime > (item.second.last_bias_s_ + bias_timeout_)))
+        {
+          // Trusted status timed out
+          item.second.bias_valid_ = false;
+        }
```

Comment:
So, the data that has timeouts would need to be periodically updated as trusted?
- Something like a heartbeat.

Reply:
That is correct, if the timeout is set. If not set, it defaults to -1, so no timeout. The idea is that a trusted state or bias information is extraordinary, so unless kept up-to-date, it shouldn't be considered the long-term position.